drivers/watchdog/wdt_npcx.c

```diff
@@ -100,10 +101,22 @@ static inline int wdt_t0out_reload(const struct device *dev)
 	/* Reload and restart T0 timer */
 	inst->T0CSR = (inst->T0CSR & ~BIT(NPCX_T0CSR_WDRST_STS)) |
 		      BIT(NPCX_T0CSR_RST);
+	/*
+	 * Wait for the T0CSR_RST bit to change from 0 to 1. This transition is expected to occur
+	 * over multiple LFCLK cycles. If a timeout occurs, we can assume that the bit has changed
+	 * from 0 -> 1 -> 0, but the polling thread missed it because it was preempted by
+	 * higher-priority threads. In this case, we can simply return.
+	 */
+	st = k_uptime_get();
+	while (!IS_BIT_SET(inst->T0CSR, NPCX_T0CSR_RST)) {
+		if (k_uptime_get() - st > NPCX_T0CSR_RST_SET_TIMEOUT) {
```

Comment:
Non-blocking question. 
The resolution of `k_uptime_get()` is based on CONFIG_SYS_CLOCK_TICKS_PER_SEC.  For NPCX, this is set to 10000, so the resolution is 100us.

Would it be better to use `k_ticks_to_us_floor32(k_uptime_ticks())` here so that this loop is only 2us instead of 100us?

Reply:
`k_uptime_get()` function returns the time in milli-seconds. 
So the loop for `NPCX_T0CSR_RST_SET_TIMEOUT` and `NPCX_T0CSR_RST_CLEAR_TIMEOUT` is 1 ms and 2 ms.
It is not 2 us.